### PR TITLE
Add maven cluster template

### PR DESCRIPTION
**Pull request by JohnNiang**
### What this PR does

As I mentioned in the title.

You can see the result:
image
image
image

/kind feature

/cc @kubesphere-sigs/sig-devops

### CODE REVIEW

File: featured/maven/maven.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    devops.kubesphere.io/displayNameZH: "Maven"
+    devops.kubesphere.io/descriptionEN: "Designed for the continuous integration of most Maven projects, including dependency downloading, testing, building, and artifact archiving."
+    devops.kubesphere.io/descriptionZH: "专为大多数 Maven 项目的持续集成而设计，包括依赖下载、测试、构建和归档。"
+    devops.kubesphere.io/icon: "maven.svg"
+spec:
+  parameters:
+    - name: CloneURL
+      description: The URL you want to clone
+      required: true
+      default: https://github.com/johnniang/maven-sample
```
Comment (LinuxSuRen) [inline featured/maven/maven.yaml]: Is that possible to use a repo which come from `kubesphere` or `kubesphere-sigs`? I highly don't recommend to use any personal repo.

File: featured/maven/maven.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  name: maven
+  annotations:
+    devops.kubesphere.io/displayNameEN: "Maven"
+    devops.kubesphere.io/displayNameZH: "Maven"
+    devops.kubesphere.io/descriptionEN: "Designed for the continuous integration of most Maven projects, including dependency downloading, testing, building, and artifact archiving."
+    devops.kubesphere.io/descriptionZH: "专为大多数 Maven 项目的持续集成而设计，包括依赖下载、测试、构建和归档。"
+    devops.kubesphere.io/icon: "maven.svg"
+spec:
+  parameters:
+    - name: CloneURL
```
Comment (LinuxSuRen) [inline featured/maven/maven.yaml]: This template only supports a git repository. How about changing the name to `GitRepository`?

File: featured/maven/maven.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      description: The URL you want to clone
+      required: true
+      default: https://github.com/johnniang/maven-sample
+    - name: CloneRevision
```
Comment (LinuxSuRen) [inline featured/maven/maven.yaml]: How about changing the name to `GitRevision`?

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: featured/maven/maven.yaml (hunk 3 above)
Comment (JohnNiang) [reply featured/maven/maven.yaml]: That's great!

**Review by JohnNiang — COMMENTED** [review]
See the result using latest template:

image
image

File: featured/maven/maven.yaml (hunk 1 above)
Comment (JohnNiang) [reply featured/maven/maven.yaml]: I've already changed the default value to https://github.com/kubesphere-sigs/demo-java.

**Review by LinuxSuRen — APPROVED** [review]

**Pull request merged**